app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -98,10 +98,22 @@ public void onResume()
         // '1' is the brightest. We attempt to maximize the brightness
         // to help barcode readers scan the barcode.
         Window window = getWindow();
-        if(window != null && settings.useMaxBrightnessDisplayingBarcode())
+        if(window != null)
         {
             WindowManager.LayoutParams attributes = window.getAttributes();
-            attributes.screenBrightness = 1F;
+
+            if (settings.useMaxBrightnessDisplayingBarcode())
+            {
+                attributes.screenBrightness = 1F;
+            }
+
+            if (settings.getKeepScreenOn())
+            {
+                window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON|
```

Comment:
From experimenting with the change on an emulator, it seems that the FLAG_KEEP_SCREEN_ON alone does the trick, even when I configured a lock screen. What do the other two flags add, and why are they necessary?

Reply:
The first flag just makes the screen not turn off automatically.

The other two flags make the lock screen not appear over the app's window, even if the user turned off the screen manually. @sphh described it quite well in #287.

Double-checking this in the docs just now, I found that [`DISMISS_KEYGUARD`](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_DISMISS_KEYGUARD) is not strictly required (_I really should stop writing code from memory!_). It only affects insecure keyguards (i.e. swipe to unlock) and causes the phone to remain unlocked when closing the window. This is, if nothing else, inconsistent with the behavior on secured devices, so I'll remove it.